src/Microsoft.DocAsCode.MarkdigEngine.Extensions/TripleColon/ImageExtension.cs

```diff
@@ -90,93 +81,88 @@ public bool TryProcessAttributes(IDictionary<string, string> attributes, out Htm
+            }
+            tripleColonObj.Attributes.TryGetValue("lightbox", out var currentLightbox); //it's okay if this is null
+            tripleColonObj.Attributes.TryGetValue("border", out var currentBorderStr); //it's okay if this is null
+            tripleColonObj.Attributes.TryGetValue("link", out var currentLink); //it's okay if this is null
+            if (!bool.TryParse(currentBorderStr, out var currentBorder))
+            {
+                if (currentType == "icon")
                 {
-                    currentType = "content";
+                    currentBorder = false;
                 }
-                tripleColonObj.Attributes.TryGetValue("lightbox", out currentLightbox); //it's okay if this is null
-                tripleColonObj.Attributes.TryGetValue("border", out currentBorderStr); //it's okay if this is null
-                tripleColonObj.Attributes.TryGetValue("link", out currentLink); //it's okay if this is null
-                if (!bool.TryParse(currentBorderStr, out currentBorder))
+                else
                 {
-                    if(currentType == "icon")
-                    {
-                        currentBorder = false;
-                    } else
-                    {
-                        currentBorder = true;
-                    }
+                    currentBorder = true;
                 }
+            }
 
-                if (currentBorder)
-                {
-                    renderer.WriteLine("<p class=\"mx-imgBorder\">");
-                } else
-                {
-                    renderer.WriteLine("<p>");
-                }
-                if (!string.IsNullOrEmpty(currentLink))
-                {
-                    var linkHtmlAttributes = new HtmlAttributes();
-                    currentLink = _context.GetLink(currentLink, obj);
-                    linkHtmlAttributes.AddProperty("href", $"{currentLink}");
-                    renderer.Write("<a").WriteAttributes(linkHtmlAttributes).WriteLine(">");
-                } else if (!string.IsNullOrEmpty(currentLightbox))
-                {
-                    var lightboxHtmlAttributes = new HtmlAttributes();
-                    var path = _context.GetLink(currentLightbox, obj);
-                    lightboxHtmlAttributes.AddProperty("href", $"{path}#lightbox");
-                    lightboxHtmlAttributes.AddProperty("data-linktype", $"relative-path");
-                    renderer.Write("<a").WriteAttributes(lightboxHtmlAttributes).WriteLine(">");
-                }
-                if(currentType != "complex")
-                {
-                    renderer.Write("<img").WriteAttributes(obj).WriteLine(">");
-                } else
-                {
-                    if (currentType == "complex" && tripleColonObj.Count == 0)
-                    {
-                        logWarning("If type is \"complex\", then descriptive content is required. Please make sure you have descriptive content.");
```

Comment:
This is an example: though this line seems stateless, `logWarning` implicitly captures [processor.LineIndex](https://github.com/dotnet/docfx/pull/6084/files#diff-1df2207e9d7ec9145d28ccc0e789985dL49) that is tied to the current processor state. When `Render` is called on a different block, there is a chance that `Line` info is mis-reported.

Reply:
Got it! Thank you for doing this, it looks much better.